pkg/profile/artifact.go

```diff
@@ -206,7 +207,16 @@ func (p *Profile) makeArtifacts(profileRepos []string, gitClient git.Git) ([]Art
 			artifacts = append(artifacts, nestedArtifacts...)
 			p.nestedName = ""
 		} else if artifact.Chart != nil {
-			helmRelease := p.makeHelmRelease(artifact, profileRepoPath)
+			var (
+				cfgMap  *corev1.ConfigMap
+				mapName string
+			)
+			if artifact.Chart.DefaultValues != nil {
+				cfgMap = p.makeDefaultValuesCfgMap(artifact.Name, *artifact.Chart.DefaultValues)
+				mapName = cfgMap.ObjectMeta.Name
+				a.Objects = append(a.Objects, cfgMap)
+			}
```

Comment:
could this maybe move to the `makeHelmRelease` func?

Reply:
hmmmm i did have it there for a minute. why did i move it? 🤔

Reply:
to troll me?

Reply:
okay i can't remember.

So i can have `makeHelmRelease` return a `[]runtime.Object` which will get added to the objects list. But it means I will have to do a `helmRelease := objects[0].(*helmv2.HelmRelease)` type thing so that we can do `helmRelease.Spec.Chart.Spec.Chart = filepath.Join....` a couple lines after.

Or I can have `makeHelmRelease` return 2 definite types, and do if nil checks around adding the cfg map to the objects list. Second one is probably nicer and less brittle i think

Reply:
> to troll me?

I am not saying that is why, I am also not saying it is not why

Reply:
Oh Gods, I'm gonna have a great time merging this. :D

Reply:
Revenge is a beautiful thing :trollface:

Reply:
oh but now what do i name this func 🤔

Reply:
> oh but now what do i name this func

`makeHelmReleaseObjects`?

Reply:
sure if you want to be boring i guess